src/utils/file_utils.gd

```diff
@@ -0,0 +1,35 @@
+class_name FileUtils
+
+var export_path = "user://"
+
+var file
+
+# on windows exports to the following path: C:\Users\{username}\AppData\Roaming\Godot\app_userdata\Fachpraktikum Multiagentensystem
+func open_writer():
+	
+	var start_time = Time.get_datetime_string_from_system().replace(":","-")
+	
+	var filename = "cell-simulation-export-" + start_time + ".csv"
+	var path = export_path + filename
+	self.file = FileAccess.open(path, FileAccess.WRITE)
+	self.file.store_csv_line(["cell_id", "cell_type", "cell_pos_x", "cell_pos_y", "antigen_code", "time"])
+
+func save_cell_state_to_csv(cell: Cell):
+	var time = Time.get_datetime_string_from_system()
+	var id = cell.get_instance_id()
+	var type = Cell.TYPES.find_key(cell.cell_state_handler.cell_type).to_lower()
+	var pos_x = cell.position.x
+	var pos_y = cell.position.y
+	var antigen_code = cell.cell_state_handler.color_code
+	if type == "macrophage" || type == "bcell" || type == "thelpercellt4":
```

Comment:
Is is superfluous since you set the `color_code` to `-1` in the `init` functions of the respective classes, or not?

Reply:
No this didnt solve it. Its probably set somewhere else.

Reply:
Then that's fine.

Reply:
The correct way would be to reset it to -1. But somewhere else its set to 0.